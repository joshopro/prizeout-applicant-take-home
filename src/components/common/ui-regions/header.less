@import '../../../utils/styles/mixins.less';
@import '../../../utils/styles/media-queries.less';

.header {
	.flexbox();
	background-color: var(--color-white);
	height: 3rem;
	padding: 0 var(--space-inner);
	border-bottom: var(--border-primary);
	
	&__logo {
		.flexbox();
		justify-self: stretch;
		align-self: stretch;

		background-size: auto 50%;
		background-position: center center;
		background-repeat: no-repeat;

		background-image: url('https://assets.prizeout.com/branding/Prizeout-Logo-Blue.svg');

		min-height: 40px;
		min-width: 160px;
		
		@media @mobile-portrait-down {
			background-image: url('https://assets.prizeout.com/branding/P-Logo-Blue.svg');
		}
	}
}
