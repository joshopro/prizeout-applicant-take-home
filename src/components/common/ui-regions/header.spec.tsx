import { render } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { Header } from './header';

describe('Test Header Component', () => {
    test('Component matches snapshot', () => {
        const { asFragment } = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });
});
