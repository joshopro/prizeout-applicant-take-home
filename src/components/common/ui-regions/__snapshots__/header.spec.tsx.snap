// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header z-index-header"
  >
    <div
      class="header__exit"
    >
      <button
        type="button"
      >
        <i
          class="far fa-sign-out fa-flip-horizontal"
        />
        <span>
          Exit
        </span>
      </button>
    </div>
    <div
      class="header__logo"
    />
    <div
      class="header__user"
    >
      <div
        class="header__quick-links"
      >
        <a
          href=""
        >
          My Giftcards
        </a>
        <a
          href=""
        >
          My Points
        </a>
      </div>
      <nav
        class="header__nav"
      >
        <button
          aria-label="Toggle"
          class="button button--primary button--small"
          data-testid="button"
          title="Toggle"
          type="button"
        >
          <span>
            Toggle
          </span>
        </button>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
